modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for SHSimpleIDListFromPath
+ * COPYRIGHT:   Copyright 2024 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <shellutils.h>
+
+enum { DIRBIT = 1, FILEBIT = 2 };
+
+static int FileStruct_Att(LPCITEMIDLIST pidl)
+{
+    return pidl && pidl->mkid.cb > 14 ? *((WORD*)((BYTE*)(pidl) + 12)) : (1 << 31); // See FileStruct in pidl.h
```

Comment:
I would suggest you just redefine `FileStruct` in this test file in order to have a more human-readable code here, instead of the obscure offset/pointer calculation.

Reply:
But FileStruct is stupid and has its most important byte defined in its parent struct and therefore is annoying to work with. This offset has been this way since 1995 and will never change.

Reply:
OK let's keep it that way then.

Reply:
Not a fan of this at all. This is not human readable.

Reply:
> This is not human readable

The attribute WORD is located 12 bytes into the FS PIDL. I don't want the struct from pidl.h because potential changes there should not be reflected here. All the bytes before the filename in a FS PIDL are always at the same location.